File: app/schemas/categories.py
from typing import List
from pydantic import BaseModel


class CategoryBase(BaseModel):
    id: int
    name: str


class CategoryCreate(BaseModel):
    name: str


class CategoryUpdate(BaseModel):
    name: str


class CategoryOut(CategoryBase):
    pass

class CategoriesOut(BaseModel):
    data: List[CategoryBase]

File: frontend/src/types/categories.ts
// src/types/category.ts
export interface CategoryBase {
  id: number;
  name: string;
}

export interface CategoryCreate {
  name: string;
}

export interface CategoryUpdate {
  name: string;
}

export interface CategoryOut extends CategoryBase {}

export interface CategoriesOut {
  data: CategoryBase[];
}
